utils/grafonnet_utils.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	JsonnetExtension  = ".libsonnet"
+	JsonnetAnnotation = "jsonnet/library"
+)
+
+func CreateFolder(configMapName, basePath string) (string, error) {
+	folderPath := fmt.Sprintf("%v/%v", basePath, configMapName)
+	_, err := os.Stat(folderPath)
+	if os.IsNotExist(err) {
+		return folderPath, os.Mkdir(folderPath, os.ModePerm)
+	}
+	return folderPath, err
+}
+
+func CreateFile(filePath, contents string) error {
+	err := validateFileExtension(filePath)
+	if err != nil {
+		return err
+	}
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	_, err = file.WriteString(contents)
+	if err != nil {
+		return err
+	}
+
+	return file.Close()
+}
+
+func validateFileExtension(filePath string) error {
+	//check for a valid jsonnet extension
+	extension := filepath.Ext(filePath)
+	if extension != JsonnetExtension {
+		return fmt.Errorf("unknown extension, expected %v", JsonnetExtension)
+	}
+	return nil
+}
```

Comment:
Will either be dropped or used in the jsonnet fetcher, i'm still deciding what a cleaner way is